iron-overlay-behavior.html

```diff
@@ -510,12 +510,25 @@
      * @protected
      */
     _onCaptureTab: function(event) {
+      if (!this.withBackdrop) {
+        return;
+      }
       // TAB wraps from last to first focusable.
       // Shift + TAB wraps from first to last focusable.
       var shift = event.shiftKey;
       var nodeToCheck = shift ? this.__firstFocusableNode : this.__lastFocusableNode;
       var nodeToSet = shift ? this.__lastFocusableNode : this.__firstFocusableNode;
-      if (this.withBackdrop && this._focusedChild === nodeToCheck) {
+      // If nodeToCheck is the same as nodeToSet, it means we have an overlay
+      // with 0 or 1 focusables; in either case we still need to trap the focus
+      // within the overlay.
+      // If the active element is not the nodeToCheck but the overlay itself,
+      // it means the focus is about to go outside the overlay, hence we should
+      // prevent that (e.g. user opens the overlay and hit Shift+TAB).
+      // In dom=shadow, the manager will receive focus changes on the main root
+      // but not the ones within other shadow roots, so we can't rely on
+      // _focusedChild, but we should check the deepest active element.
+      if (nodeToCheck === nodeToSet ||
+          [nodeToCheck, this].indexOf(this._manager.deepActiveElement) !== -1) {
```

Comment:
This is an unnecessarily fancy way to compare two things to another thing. I recommend sticking to normal equality checks and boolean operators.